### PR TITLE
MYST-18 define client promise

### PR DESCRIPTION
**Pull request by interro**

### CODE REVIEW

File: client_promise/dto/promise.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package dto
+
+type ClientPromise struct {
+	SerialNumber int
+	IssuerId int
+	BenefiterId int
+	Amount int
```
Comment (Waldz) [inline client_promise/dto/promise.go]: - Use type from `service_discovery/dto/price.go`
- Rename `Price` -> `Money`

File: client_promise/dto/promise.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package dto
+
+type ClientPromise struct {
```
Comment (Waldz) [inline client_promise/dto/promise.go]: - Create structure `type Signature string`
- Add field `issuerSignature *Signature`, test if is `nil` by default
- Add field `BenefiterSignature` *Signature`, test if is `nil` by default

File: client_promise/promise.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
+  "promise": {
+    "id": 1,
```
[the remark this answers is not in the document]
Comment (cirka) [reply client_promise/promise.json]: It should be  serial_number +

File: client_promise/promise.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    "proposal_id": 1,
+    "issuer_id": 1,
+    "amount": 5,
```
[the remark this answers is not in the document]
Comment (cirka) [reply client_promise/promise.json]: amount of tokens promised is scalar.

File: client_promise/promise.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+    "date_created": "unix_timestamp"
```
[the remark this answers is not in the document]
Comment (cirka) [reply client_promise/promise.json]: We do not need a timestamp in a promise structure at all.

File: service_discovery/dto/money.go [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package dto
+
+type Money struct {
+	Amount   uint64
```
Comment (Waldz) [inline service_discovery/dto/money.go]: Some tests probably broke, check by running `bin/test`
